civihr_employee_portal/js/reports.js

```diff
@@ -331,13 +323,51 @@
     }
 
     /**
+     * Init angular in reports custom page,
+     * and add the calendar icon
+     *
+     */
+    HRReport.prototype.initAngular = function() {
+        var self = this;
```

Comment:
Avoid using the `self` keyword. Modern browsers provide a global `self` [variable](https://developer.mozilla.org/en-US/docs/Web/API/Window/self) , and that can cause confusion. I've seen people call it `that`